src/pg/replication.cr

```diff
@@ -0,0 +1,353 @@
+module PG::Replication
+  module Handler
+    def received(frame)
+    end
+  end
+
+  class Connection
+    getter handler : Handler
+    getter publication_name : String
+    getter slot_name : String
+
+    def initialize(uri : URI | String, @handler, *, @publication_name, @slot_name, blocking : Bool = false)
+      if uri.is_a? String
+        uri = URI.parse(uri)
+      else
+        uri = uri.dup
+      end
+      query_params = uri.query_params
+      query_params["replication"] = "database"
+      uri.query_params = query_params
+      @conn = DB.connect(uri).as(PG::Connection)
+      @conn.listen_replication(
+        publication_name: publication_name,
+        slot_name: slot_name,
+        blocking: blocking,
+      ) do |frame|
+        received frame
+      end
+    end
+
+    def received(frame : CopyBoth)
+    end
```

Comment:
I don't know what to do here or if there's anything _to_ do. This is the first message that comes in after you send the `START_REPLICATION` command.